File: benchmarks/fibonacci/naive.c

#include "../../timing/dphpc_timing.h"

long fib(int n) {
    if (n <= 2) {
        return 1;
    }
    return fib(n-1) + fib(n-2);
}

int main() {
    dphpc_time(
        fib(35);
    );
}

File: benchmarks/fibonacci/naive.jl


include("../../timing/dphpc_timing.jl")

fib(n) = n <= 2 ? 1 : fib(n-1) + fib(n-2)

function main()

    result = @dphpc_time fib(38)

    return "blub" # (no longer need to return result 🙃)
end

main()